### PR TITLE
ci: configure eslint workflow

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR adds the CI workflows for ESLint and build steps.

### CODE REVIEW

File: .github/workflows/build.yaml
Comment (drin) [inline .github/workflows/build.yaml:1]: Nice to use npm LTS version.

**Review by drin — APPROVED** [review]
I have not setup build actions for CI before, but what I see in the build config seems fine to me.

It may be overkill, but the [starter workflow](https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs#using-the-nodejs-starter-workflow) has a few things that I kind of like, specifically:
- includes node version in the step name
- specifies branches to trigger on (maybe we actually want it on all branches?)
- runs tests (even if we don't have any, maybe it's good to just have it there?)

None of the above is necessary though, and I think this looks fine.

**Review by drin — COMMENTED** [review]
I think the lint action looks good.

Looking at a [starter workflow for ESLint](https://github.com/actions/starter-workflows/blob/f3c5d7931d054ffbbdcbfdc463cc3bd0def74929/code-scanning/eslint.yml#L21-L50), it seems it may be more flexible to npm install eslint and then run it with npx, though I guess defining the lint command in `scripts` works fine too (especially since this is a simple project so far).

Also, I wonder if there's a reason not to use `actions/checkout@v3` or maybe it doesn't matter?

**Review by drin — APPROVED** [review]

**Review by vbarua — COMMENTED** [review]
Hello! I think this looks like a perfectly good start for this.

@sanjibansg would you be able to run this action in your fork? I was looking at https://github.com/sanjibansg/substrait-js/actions because I wanted to see at least one success before merging this in.

File: .github/workflows/build.yaml
Comment (sanjibansg) [reply .github/workflows/build.yaml:1]: Thanks for the review.

- I might be missing something here, but we do have the node version mentioned here,
https://github.com/substrait-io/substrait-js/pull/4/files#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1R14
- The workflow right now specifies running the build and lint checks on every push and pull requests.
- In PR #8 I am introducing tests, so will add this change there.

**Comment by sanjibansg** [comment]
> Hello! I think this looks like a perfectly good start for this.
> 
> @sanjibansg would you be able to run this action in your fork? I was looking at https://github.com/sanjibansg/substrait-js/actions because I wanted to see at least one success before merging this in.

Hi @vbarua 
Thanks for the review!
The CI checks now run successfully here https://github.com/sanjibansg/substrait-js/actions

**Review by vbarua — APPROVED** [review]
Thanks for adding this and getting the check to pass 🙇

**Pull request merged**